src/TTextEdit.cpp

```diff
@@ -549,34 +549,32 @@ int TTextEdit::drawGrapheme(QPainter& painter, const QPoint& cursor, const QStri
     painter.drawText(textRect.x(), textRect.bottom() - mFontDescent, grapheme);
     return charWidth;
 }
+
 int TTextEdit::getGraphemeWidth(uint unicode) const
 {
-    // mk_wcwidth returns -1 (on error), 0 on a control or combining
-    // diacritical codepoint or 1 for a normal character or 2 on a wide
-    // character.
-    // mk_wcwidth_cjk does the same except it returns 2 instead of 1 for
-    // characters that have an "ambiguous" East Asian width:
-    // replaced by wide_wcwidth which returns 1 or 2 or a number of
-    // (negative) special values:
-    // mWideAmbigousWidthGlyphs
+    // Markus Kuhn's mk_wcwidth()/mk_wcwidth_cjk():
+    // https://www.cl.cam.ac.uk/~mgk25/ucs/wcwidth.c have not been updated since
+    // Unicode 5 and we have replaced them by wide_wcwidth:
+    // https://github.com/ridiculousfish/widecharwidth
+    // which returns 1 or 2 or a number of (negative) special values:
     switch (widechar_wcwidth(unicode)) {
     case 1: // Draw as normal/narrow
         return 1;
     case 2: // Draw as wide
         return 2;
     case widechar_nonprint:     // -1 = The character is not printable.
-        qDebug().nospace().noquote() << "TTextEdit::drawGrapheme(...) WARN - trying to draw a Unicode character which is unprintable, codepoint number: U+" << QString::number(unicode, 16) << ".";
+        qDebug().nospace().noquote() << "TTextEdit::getGraphemeWidth(...) WARN - trying to get width of a Unicode character which is unprintable, codepoint number: U+" << QString::number(unicode, 16) << ".";
```

Comment:
fyi this message spams the hell out of you when you are working with emojis, suggest dropping it. Unrelated to this PR however.